pkg/archive/archive_test.go

```diff
@@ -724,6 +725,77 @@ func TestTarUntar(t *testing.T) {
 	}
 }
 
+func TestTarWithOptionsChownOptsAlwaysOverridesIdPair(t *testing.T) {
+	origin, err := ioutil.TempDir("", "docker-test-tar-chown-opt")
+	require.NoError(t, err)
+
+	defer os.RemoveAll(origin)
+	filePath := filepath.Join(origin, "1")
+	// unprivilegeduser idPair
+	originalUID := 1000
+	originalGID := 999
+	err = ioutil.WriteFile(filePath, []byte("hello world"), 0700)
+	os.Chown(filePath, originalUID, originalGID)
+	require.NoError(t, err)
+
+	idMaps := []idtools.IDMap{
+		0: {
+			ContainerID: 0,
+			HostID:      0,
+			Size:        65536,
+		},
+		1: {
+			ContainerID: 0,
+			HostID:      100000,
+			Size:        65536,
+		},
+	}
+
+	// Initial case, without chownOpts the idPair is the one from the host file
```

Comment:
I don't think it is needed here. If we want to test it, it should be part of the regular `TarWithOptions` test that doesn't pass parameters.